clang/lib/AST/DeclPrinter.cpp

```diff
@@ -873,9 +873,14 @@ void DeclPrinter::VisitFunctionDecl(FunctionDecl *D) {
 
   if (D->isPureVirtual())
     Out << " = 0";
-  else if (D->isDeletedAsWritten())
+  else if (D->isDeletedAsWritten()) {
     Out << " = delete";
-  else if (D->isExplicitlyDefaulted())
+    if (const auto *M = D->getDeletedMessage()) {
```

Comment:
Could you please spell out the type?
```suggestion
    if (const StringLiteral *M = D->getDeletedMessage()) {
```

Reply:
Ah yeah, I have a habit of always writing `auto`, so I forget about this sometimes; will do.